pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -579,7 +579,7 @@ cdef class _Timestamp(ABCTimestamp):
         if freq:
             kwds = freq.kwds
             month_kw = kwds.get("startingMonth", kwds.get("month", 12))
-            freqstr = freq.freqstr
+            freqstr = to_offset(freq.freqstr).name
```

Comment:
does it work to directly do `freq.name`?

Reply:
thanks, it works with `freq.name` indeed. I simplified `to_offset(freq.freqstr).name`